scripts/quality/run_codex_exec.py

```
@@ -96,13 +115,17 @@ def build_codex_command(args: argparse.Namespace) -> List[str]:
+    args: argparse.Namespace, prompt_text: str
+) -> subprocess.CompletedProcess:
     """Run codex with a static literal argv list and prompt text passed via stdin."""
     executable_path = _resolved_codex_executable_path()
     command = build_codex_command(args)
-    # Safe-by-construction: a fixed literal executable name, explicit argv, shell=False,
-    # validated tokens as plain arguments, and prompt content flowing only through stdin.
-    return subprocess.run(  # nosec B603  # nosemgrep: python.lang.security.audit.dangerous-subprocess-use-audit.dangerous-subprocess-use-audit
+    # Safe-by-construction: a fixed literal executable name, explicit argv,
+    # shell=False, validated tokens as plain arguments, and prompt content
+    # flowing only through stdin.
+    return subprocess.run(  # nosec B603
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)